### ISSUE TITLE
Bert: Attention mask is required for embedding calculation on BGE model

### ISSUE THREAD

**Issue by lucasavila00** [huggingface/candle#1798]
Attention mask, as already reported here https://github.com/huggingface/candle/issues/1552 is not available.

However, that issue was closed because bert is used only for computing embeddings.

But the attention mask is required for proper calculation of many sentences at once, otherwise, results will not be correct.

I created a repository that reproduces it. https://github.com/lucasavila00/candlerepro

Main Logs: https://github.com/lucasavila00/candlerepro/blob/main/log.log

Cheking the logs of the main function shows us that if we process one item at a time, we get the right results.
Also, it shows that my implementation for batching (basically copied from the examples) adds the right padding.
And it shows that the batched implementation, if using just one item, is also correct.

However, if 2 items are used, the paddings will affect results due to missing attention mask.

The results were also checked against this Python implementation: https://github.com/lucasavila00/candlerepro/blob/main/test.ipynb

It shows all values match, however the tokenizer in Python creates the required attention mask and passes it to the model call.

**Comment by lucasavila00** [comment]
I'm reading trough transformers' code and it seems what is required for the correct calculation is this extended attention mask https://github.com/huggingface/transformers/blob/6eedfa6dd15dc1e22a55ae036f681914e5a0d9a1/src/transformers/modeling_utils.py#L877C1-L910C39

Where it is `extended_attention_mask = (1.0 - extended_attention_mask) * torch.finfo(dtype).min`, the rest of the function does broadcasting.

Then, this needs to be added to the attention scores, like here https://github.com/huggingface/transformers/blob/6eedfa6dd15dc1e22a55ae036f681914e5a0d9a1/src/transformers/models/bert/modeling_bert.py#L350-L352

The tokenizer also needs to output the embedding mask.


This seems like a reasonably simple change, I'll try creating a PR
